### ISSUE TITLE
ArgumentException: An item with the same key has already been added. in HuaweiStore.ParseProducts

### ISSUE THREAD

**Issue by AntonPetrov83** [EvilMindDevs/hms-unity-plugin#385]
```
ArgumentException: An item with the same key has already been added. Key: bundle_starterpack
   at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior)
   at HmsPlugin.HuaweiStore.ParseProducts (HuaweiMobileServices.IAP.ProductInfoResult result, HuaweiMobileServices.IAP.PriceType type)
   at HmsPlugin.HuaweiStore+<>c__DisplayClass19_0.<CreateProductRequest>b__0 (HuaweiMobileServices.IAP.ProductInfoResult result)
   at System.Action`1[T].Invoke (T obj)
   at System.Action.Invoke ()
   at HuaweiMobileServices.Utils.HMSDispatcher.Update ()
```

Caling `UnityPurchasing.Initialize()` multiple times is not prohibited and this is what our game does when a game session is restarted.

Calling `Initialize()` second time leads to a  `LoadComsumableProducts()` being called twice.
Then `ArgumentException: An item with the same key has already been added.` is being thrown from `ParseProducts()` method.

I think it can be fixed by seting `clientInited = false;` in `IStore.Initialize()`

**Comment by alihan98ersoy** [comment]
I found some solutions on the internet. Maybe these can help.

1. https://stackoverflow.com/a/5648119
Most likely, you have a model which contains the same property twice. Perhaps you are using new to hide the base property.
The solution is to override the property or use another name.
If you share your model, we would be able to elaborate more.

2. https://stackoverflow.com/a/26516900
This error is fairly self-explanatory. Dictionary keys are unique and you cannot have more than one of the same key. To fix this, you should modify your code like so:

3. https://forum.uipath.com/t/assign-an-item-with-the-same-key-has-already-been-added/408578

**Comment by AntonPetrov83** [comment]
@alihan98ersoy Hi! Of course I know what this exception is about. I described why it happens and how it can be fixed.
I did the fix in my copy of the HMS plugin.

P.S. Great plug-in btw!

**Comment by alihan98ersoy** [comment]
Ohh I see. Thank you

**Comment by eozdemir90** [comment]
Thank you for your interest in our project. You may give a star to support us and follow up on the project. You can also join our discord channel to follow changes and ask any kind of question about the plugin, gaming ecosystem, etc. You may access it from here : https://discord.io/hmsunityplugin
Thank you and stay tuned!

**Comment by bunyamineymen** [comment]
Hi,
Our plugin has a demo using IAP via HuaweiStore.cs anymore.We have just released new version 2.2.8.
